test/server/generators.ts

```diff
@@ -0,0 +1,51 @@
+import { expect, test } from 'vitest'
+import { app } from './utils'
+
+test('typescript generator route', async () => {
```

Comment:
not actually testing anything

Reply:
I would actually disagree, even if minimal this ensure that the route doesn't throw an exception (which occurs if invalid TS is generated when passing within prettier for instance, or if there is a bug in the JS within the string interpolations used to generate the content).

That's not a great test, but that's a start.

Reply:
and if due to code changes it this route starts to return no types instead of an error, can you spot it with this test?

this test can be sort of easily improved if you have a file with types saved somewhere and you compare the body with the template file (it can even be contains and some expected parts)

Reply:
Yeah, that's not great. I guess in the case you mention "public" will probably not be present in the "schemas" hence fail the test. But I agree that's brittle.

Looking at your suggestion this might be a good place to use `toMatchInlineSnapshot` so this can still be updated easily if types generation changes and make it more easily spotable in reviews.

Reply:
we can just create a table, a function, a view, etc in the beginning of the test and check that types are generated as expected for these